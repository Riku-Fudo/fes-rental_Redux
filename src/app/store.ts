import { configureStore } from '@reduxjs/toolkit'

import loginSlice from '../features/loginSlice'
import itemsSlice from '../features/itemsSlice'

export const store =  configureStore({
  reducer: {
    login: loginSlice,
    items: itemsSlice
  }
})
export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
