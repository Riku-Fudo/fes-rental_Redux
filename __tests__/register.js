'use strict';

import renderer from 'react-test-renderer';
import LoginScreen from '../src/pages/register';
import { Provider } from 'react-redux'
import { store } from '../src/app/store'

// モック用に追記
jest.mock('next/router', () => ({
    useRouter() {
      return {
        asPath: '/registerComp',
      };
    },
  }));

describe('next-router-mock', () => {
    it('renders loginPages', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <LoginScreen></LoginScreen>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
});
