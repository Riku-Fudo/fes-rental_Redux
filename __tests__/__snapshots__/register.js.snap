// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`next-router-mock renders loginPages 1`] = `
[
  <header
    className="header"
  >
    <div
      className="info"
    >
      <img
        alt="タイトルロゴ"
        data-nimg="1"
        decoding="async"
        height={70}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Flogo.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={232}
      />
    </div>
  </header>,
  <main
    className="registMain"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <ul>
        <li>
          <div
            className="labelWrapper"
          >
            <label
              className="label"
              htmlFor="userName"
            >
              ユーザー名
            </label>
            <span
              className="required"
            >
              必須
            </span>
          </div>
          <input
            id="userName"
            name="userName"
            onChange={[Function]}
            type="text"
            value=""
          />
        </li>
        <li>
          <div
            className="labelWrapper"
          >
            <label
              className="label"
              htmlFor="zipcode"
            >
              郵便番号
            </label>
            <span
              className="required"
            >
              必須
            </span>
          </div>
          <input
            id="zipcode"
            name="zipcode"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="Search"
            id="btn-search"
            onClick={[Function]}
            type="button"
          >
            住所検索
          </button>
        </li>
        <li
          className="listWrapper"
        >
          <div
            className="listflex"
          >
            <div
              className="listInfo"
            >
              <div
                className="labelWrapper"
              >
                <label
                  className="label"
                  htmlFor="prefectures"
                >
                  都道府県
                </label>
                <span
                  className="required"
                >
                  必須
                </span>
              </div>
              <input
                id="prefectures"
                name="prefectures"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="listInfo"
            >
              <div
                className="labelWrapper"
              >
                <label
                  className="label"
                  htmlFor="city"
                >
                  市区町村
                </label>
                <span
                  className="required"
                >
                  必須
                </span>
              </div>
              <input
                id="city"
                name="city"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="listflex"
          >
            <div
              className="listInfo"
            >
              <div
                className="labelWrapper"
              >
                <label
                  className="label"
                  htmlFor="houseNumber"
                >
                  番地
                </label>
                <span
                  className="required"
                >
                  必須
                </span>
              </div>
              <input
                id="houseNumber"
                name="houseNumber"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="listInfo"
            >
              <div
                className="labelWrapper"
              >
                <label
                  className="label"
                  htmlFor="buildingName"
                >
                  建物名・部屋番号
                </label>
              </div>
              <input
                id="buildingName"
                name="buildingName"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </li>
        <li
          className="listWrapperFlex listWrapper"
        >
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="familyName"
              >
                姓
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="familyName"
              name="familyName"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="firstName"
              >
                名
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="firstName"
              name="firstName"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </li>
        <li
          className="listWrapperFlex listWrapper"
        >
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="familyNameKana"
              >
                セイ
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="familyNameKana"
              name="familyNameKana"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="firstNameKana"
              >
                メイ
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="firstNameKana"
              name="firstNameKana"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </li>
        <li
          className="listWrapper"
        >
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="phoneNumbe"
              >
                電話番号
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="phoneNumbe"
              name="phoneNumbe"
              onChange={[Function]}
              type="tel"
              value=""
            />
          </div>
        </li>
        <li>
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="mailAddress"
              >
                メールアドレス
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="mailAddress"
              name="mailAddress"
              onChange={[Function]}
              type="email"
              value=""
            />
          </div>
        </li>
        <li
          className="listWrapperFlex listWrapper"
        >
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="password"
              >
                パスワード
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="password"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="listInfo"
          >
            <div
              className="labelWrapper"
            >
              <label
                className="label"
                htmlFor="passwordTest"
              >
                確認用パスワード
              </label>
              <span
                className="required"
              >
                必須
              </span>
            </div>
            <input
              id="passwordTest"
              name="passwordTest"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
        </li>
      </ul>
      <div
        className="btnWrapper"
      >
        <button
          className="registBtn"
          type="submit"
        >
          登録する
        </button>
      </div>
      <div
        className="linkWrapper"
      >
        <a
          className="topLink"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          会員登録せずトップページへ
        </a>
      </div>
    </form>
  </main>,
]
`;
